src/main/software/amazon/event/ruler/input/ByteParser.java

```diff
@@ -0,0 +1,10 @@
+package software.amazon.event.ruler.input;
+
+public interface ByteParser {
```

Comment:
I'd recommend you also include a minor doc on what we expect to use this interface. nothing too fancy, but enough to offer direction for anyone that comes across this class.